### ISSUE TITLE
Feature questions

### ISSUE THREAD

**Issue by Jessime** [phn/lineid_plot#4]
Hi,

I've got a couple of questions about lineid_plot; hopefully this is an appropriate place to ask.
 
1. Is it possible to specify the color of the text? It looks like it isn't, but I think it should be pretty straightforward to add color as a kwarg, right?
2. How difficult would it be to apply this algorithm to the y-axis of a plot? I haven't completely digested the code, but it looks like you'd basically have to write a whole separate function.

This script is almost exactly what I've been looking for. I need some way to apply labels to the y-axis of an image, and all of my data is already in Python. If the y-axis issue is as much as an overhaul as I'm guessing, do you have any idea if there's anything along these lines already available? 

Thanks!

**Comment by phn** [comment]
Hello,

1.

Yes it would be pretty easy to add text color to ``kwargs`` in ``plot_line_ids``. But I wanted the function to accept only items necessary for the layout. I would recommend writing a simple wrapper function that first calls ``plot_line_ids`` and then uses the ``axes`` instance returned from ``plot_line_ids`` to change the text color.

See https://gist.github.com/phn/2326396 for examples of such functions.

2.

Applying the algorithm to y-axis, that is calculating the position where labels should be placed, should be straight forward, I think. The function that calculates the position of labels is ``adjust_boxes``. This function doesn't know anything about axis orientation other that that it has "left" and "right" edges, which should translate to "lower" and "upper" edges. My guess is that this should work either as is or with minor modification on the y-axis. 

Getting the main function, ``plot_line_ids``, to work with y-axis would need a bit more work, as it assumes that we are dealing with x-axis. 

----

I do not know of any other script like this. My recommendation would be to first try to get ``adjust_boxes`` to work with y-axis. That is, give it y-axis "boxes" and see what happens. Make adjustments, most likely minor, to fix any mistakes. Then adapt the code in ``plot_line_ids``, perhaps wraping the relevant code in ``if`` statement to handle y-axis and x-axis separately.

Of-course the devil is in the details. I will try to look into this during the weekend. But if you have time to look into this do give it a try and let me know what you think.

Hope this helps.

Regards,
Prasanth

**Comment by Jessime** [comment]
Thanks for the reply!

I got everything I wanted worked out. It's probably a little bit hacky, but I'm using lineid_plot and the `color_text_boxes` function from your gist to automatically generate some nice labels for some heatmaps I have. I ended up realizing that I could just transpose my matrix instead of switching the labels to the yaxis, haha. I can share the code if you're interested, but like I mentioned, it isn't very pretty at the moment.

Nice job on this code and thanks again for the help!
